sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVectorCore.java

```diff
@@ -159,7 +161,9 @@ public final ColumnarMap getMap(int ordinal) {
 
   @Override
   public final Decimal getDecimal(int rowId, int precision, int scale) {
-    throw new IllegalStateException("The decimal type is currently not supported by rapids cudf");
+    BigDecimal bigDec = cudfCv.getBigDecimal(rowId).setScale(scale, RoundingMode.UNNECESSARY);
+    assert bigDec.precision() <= precision : "Assert " + bigDec.precision() + " <= " + precision;
+    return Decimal.apply(bigDec);
```

Comment:
This is performance sensitive, since it's called by the columnar to row conversion.  As such it would be great to avoid the full `BigDecimal` treatment if possible.  AFAICT the scale being requested here should always match the underlying scale being used.  Should we do something like the following to avoid `BigDecimal` in what I believe will be the common (only?) case in practice?
```suggestion
    if (precision <= DType.DECIMAL64_MAX_PRECISION && scale == cudf.getType().getScale()) {
      assert cudfCv.getType().getTypeId() == DTypeEnum.DECIMAL64;
      return Decimal.createUnsafe(cudfCv.getLong(rowId), precision, scale);
    }
    BigDecimal bigDec = cudfCv.getBigDecimal(rowId).setScale(scale, RoundingMode.UNNECESSARY);
    assert bigDec.precision() <= precision : "Assert " + bigDec.precision() + " <= " + precision;
    return Decimal.apply(bigDec);
```

Reply:
I am fairly confident that the scale/precision will always match. So could we just turn it into assertions instead of an if/else? I would like to avoid conditionals if possible too.

```
assert precision <= DType.DECIMAL64_MAX_PRECISION : ...
assert cudf.getType().getTYpeId() == DTypeEnum.DECIMAL64 : ....
assert scale == cudf.getType().getScale() : ....
return Decimal.createUnsafe(cudfCv.getLong(rowId), precision, scale);
```

Reply:
Also I forgot to add that once this goes in and we have integration test support for decimal I will update the fast path to support decimal too.

Reply:
Sounds great.  I'm all for removing the `BigDecimal` fallback code if we're confident that will never be necessary.  Actually I think we can also remove the `assert cudf.getType().getTYpeId() == DTypeEnum.DECIMAL64` I proposed since `cudfCv.getLong` will itself assert that the type is backed by a long.